pymodbus/client/mixin.py

```diff
@@ -708,8 +708,12 @@ def convert_from_registers(
         for x in registers:
             byte_list.extend(int.to_bytes(x, 2, "big"))
         if data_type == cls.DATATYPE.STRING:
-            if byte_list[-1:] == b"\00":
-                byte_list = byte_list[:-1]
+            # remove trailing null bytes
+            trailing_nulls_begin = len(byte_list)
+            while trailing_nulls_begin > 0 and byte_list[trailing_nulls_begin - 1] == 0:
```

Comment:
You compare an int with a byte, this forces a conversion, please change to compare to a byte.